src/ape_ethereum/provider.py

```diff
@@ -624,6 +624,25 @@ def get_receipt(
         )
         hex_hash = HexBytes(txn_hash)
 
+        txn = {}
+        if transaction := kwargs.get("transaction"):
+            # perf: If called `send_transaction()`, we should already have the data!
+            txn = (
+                transaction
+                if isinstance(transaction, dict)
+                else transaction.model_dump(by_alias=True, mode="json")
+            )
+
+        if kwargs.get("private"):
+            # Bail before confirmation because it won't be on chain yet.
+            data = {
+                "required_confirmations": 0,
+                "block_number": -1,
+                **txn,
```

Comment:
**problem**: we are going to need access to `timeout` for properly timing out during the potential manual-call to wait for tx acceptance, so maybe plop that in the kwargs here? I am not sure if it is being used anywhere but we will have to figure that out in the core refactor.

if it causes problems, we should add a TODO to remember that because knowing the calculated timeout is Alchemy-specific and the receipt should have that.